src/test/java/com/game/racing/car/CarsTest.java

```diff
@@ -0,0 +1,79 @@
+package com.game.racing.car;
+
+import com.game.racing.generator.MovableIntegerGenerator;
+import com.game.racing.generator.NotMovableIntegerGenerator;
+import com.game.racing.generator.RandomIntegerGenerator;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class CarsTest {
+
+    private Cars cars;
+
+    /**
+     * 전체 자동차가 RandomIntegerGenerator 를 사용하여 움직이는 경우
+     * 움직인 자동차 수가 전체 자동차 수 이하임을 확인한다.
+     */
+    @Test
+    public void cars_moveCars_byRandomIntegerGenerator() {
+        Integer carCount = 3;
+        cars = new Cars(carCount, new RandomIntegerGenerator());
+        Assertions.assertThat(cars.moveCars()).isLessThanOrEqualTo(carCount);
+    }
+
+    /**
+     * 전체 자동차가 MovableIntegerGenerator 를 사용하여 움직이는 경우
+     * 움직인 자동차 수가 전체 자동차 수와 동일한지 확인한다.
+     */
+    @Test
+    public void cars_moveCars_byMovableIntegerGenerator() {
+        Integer carCount = 3;
+        cars = new Cars(carCount, new MovableIntegerGenerator());
+        Assertions.assertThat(cars.moveCars()).isEqualTo(carCount);
+    }
+
+    /**
+     * 전체 자동차가 NotMovableIntegerGenerator 를 사용하여 움직이는 경우
+     * 움직인 자동차 수가 없는지 확인한다.
+     */
+    @Test
+    public void cars_moveCars_byNotMovableIntegerGenerator() {
+        Integer carCount = 3;
+        cars = new Cars(carCount, new NotMovableIntegerGenerator());
+        Assertions.assertThat(cars.moveCars()).isEqualTo(0);
```

Comment:
```suggestion
        Assertions.assertThat(cars.moveCars()).isZero();
```
이런식으로도 작성이 가능합니다. 어떤것이 더 가독성이 높은지는 개인취향일듯 합니다 ㅎㅎ

Reply:
아 이런 메소드도 있군요? 감사합니다! 정말 다양하네요 Assert ! ㅋㅋ 내용 반영하겠습니다 👍